std/traits.d

```diff
@@ -1480,6 +1484,180 @@ private FunctionAttribute extractAttribFlags(Attribs...)()
     return res;
 }
 
+/**
+Checks whether a function has the given attributes attached.
+
+Params:
+    func = function to check
+    attributes = variadic number of function attributes
+
+Returns:
+    `true`, if the function has the list of attributes attached and `false` otherwise.
+
+See_Also:
+    $(LREF functionAttributes)
+*/
+template hasFunctionAttributes(func...)
+    if (isCallable!(func[0]) && func.length > 0)
+{
+    enum bool hasFunctionAttributes = {
+        enum funcAttribs = __traits(getFunctionAttributes, func[0]);
+        mainloop: foreach (attribute; func[1..$])
+        {
+            foreach (attrib; funcAttribs)
+            {
+                static if (attrib == attribute)
+                    continue mainloop;
```

Comment:
Not sure if this works, but it would be nicer:

     if (funcAttribs.canFind(attribute)) continue;